memory-cards.js

```diff
@@ -1,19 +1,24 @@
 var MemoryCards = (function() {
```

Comment:
Looks good! I've tried to answer the specific questions you asked about problem 3, but let me know if there's anything else.

Oh, and don't forget to do the work in a sub-folder rather than in the main folder! If this gets merged in, it will overwrite the original problems.